### ISSUE TITLE
module_starter

### ISSUE THREAD

**Issue by christoph-morrison** [fhem/mod_template#1]
Pointer auf https://github.com/christoph-morrison/Maschinengeist-Utils-module_starter

**Comment by sidey79** [comment]
Sieht durchaus interessant aber auch komplex aus.

Ich nehme an, das ist ein Script, welches ein Modul generiert und diverse Verzeichnisse anlegt?

Ich wollte hier ein Template Repository erzeugen in dem die nötige Struktur und Hilfsmittel vor konfiguriert sind.
Habe mich auf die schnelle dem Hello World Beispiel aus dem wiki bedient und es ein wenig angereichert:-)

**Comment by christoph-morrison** [comment]
> Sieht durchaus interessant aber auch komplex aus.
> Ich nehme an, das ist ein Script, welches ein Modul generiert und diverse Verzeichnisse anlegt?

Genau. Das Skript orientiert sich dabei am CPAN-Module-Starter und soll den Autor von ein paar Basisdingen entlasten, die eigentlich jedes Modul hat (Attribut-, Set- und Get-Behandlung, Define, etc.).

> Ich wollte hier ein Template Repository erzeugen in dem die nötige Struktur und Hilfsmittel vor konfiguriert sind.
> Habe mich auf die schnelle dem Hello World Beispiel aus dem wiki bedient und es ein wenig angereichert:-)

Das kann sich ja ergänzen. Ich hab z.B. keine Zeile GH Actions oder so drin.

**Comment by sidey79** [comment]
Ja, das denke ich auch, dass sich beides ergänzen könnte, aber ich weiss noch nicht genau wie.

Die Idee vom Starter ist ja, dass es vom Modulersteller ausgeführt wird.
Ich nehme an, es ist für eine interaktive Sitzung gedacht um die Parameter abzufragen. Das ließe sich zwar grundsätzlich auch durch GitHub actions eingeben, aber der Modulersteller müsste wissen wie und wo er die nötigen Parameter angibt.
Alles lösbar, aber vielleicht auch zu kompliziert?

**Comment by sidey79** [comment]
Ich hab das Startermodul ausprobiert und die Parameter ein wenig durch try & Error getestet.

Vermutlich liegt es an der Angabe der Parameter, ich habe es leider nicht geschafft, dass im Ordner FHEM ein Fhem Modul angelegt wird und im Ordner Lib de gleichen Ebene ein Package.

Meiner Auffassung nach müsste ja die Ordnerstruktur so sein:

```
/ 
/FHEM
              /20_MyModule.pm
/lib
        /MyPackage/MyPackage.pm


/t
    ....
```

Den Einsatz von Readonly sehe ich leider auch ein bisschen problematisch.
Ich würde es auf der einen Seite sehr begrüßen, wenn Readonly standartmäßig mit FHEM installierte werden würde, ich glaube aber im Debian Paket ist das derzeit noch nicht als Abhängigkeit hinterlegt.
Das führt dann bei unerfahrenen dazu, dass das neue Modul nicht läuft :)

Vielleicht ist das aber auch nicht die Zielgruppe, ich weiss nicht so genau.

Bin unabhängig davon seit ein paar Tagen am überlegen, ob sich da ein kleine Hilfsroutine anbieten würde, die einem das Prüfen auf Vorhandensein erleichtert.
